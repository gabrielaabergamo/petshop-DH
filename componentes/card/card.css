.card-retangulo{
    width: 260px;
    height: 250px;
    background-color: white;
}

.card-info {
    margin-top: 15px;
    margin-bottom: 45px;
}

.card-titulo{
    font-weight: 800;
}
.card-preco{
    font-size: 15px;
}