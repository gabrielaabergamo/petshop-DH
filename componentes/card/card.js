import { formatter } from '../../js/formatmoney';
import { getData } from '../../js/db-request';

export const createMarkupCard = async () => {
  const data = await getData();

  return data.map((servico) => `
    <div class="card">
        <div class="card-retangulo"></div>
        <div class="card-info">
            <span class="card-titulo">${servico.title}</span> <br>
            <small class="card-preco">${formatter.format(servico.price)}</small>
        </div>
    </div>
    `).join('');
};
