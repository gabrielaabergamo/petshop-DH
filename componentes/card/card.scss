.card{
    &-retangulo{
        width: 260px;
        height: 250px;
        background-color: $background-primary-color;
    }
    &-info{
        margin-top: 15px;
        margin-bottom: 45px;
    }
    &-titulo{
        font-weight: 800;
    }
    &-preco{
        font-size: 15px;
    } 
}