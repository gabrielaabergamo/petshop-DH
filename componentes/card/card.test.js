import {createMarkupCard} from "./card"

test("createMarkupCard should render cards", ()=>{
    expect(createMarkupCard()).toMatchSnapshot()
})