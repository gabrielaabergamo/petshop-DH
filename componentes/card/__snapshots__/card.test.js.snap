// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createMarkupCard should render cards 1`] = `
"
    <div class=\\"card\\">
        <div class=\\"card-retangulo\\"></div>
        <div class=\\"card-info\\">
            <span class=\\"card-titulo\\">banho completo</span> <br>
            <small class=\\"card-preco\\">R$ 40,00</small>
        </div>
    </div>
    
    <div class=\\"card\\">
        <div class=\\"card-retangulo\\"></div>
        <div class=\\"card-info\\">
            <span class=\\"card-titulo\\">banho e tosa</span> <br>
            <small class=\\"card-preco\\">R$ 60,00</small>
        </div>
    </div>
    
    <div class=\\"card\\">
        <div class=\\"card-retangulo\\"></div>
        <div class=\\"card-info\\">
            <span class=\\"card-titulo\\">consulta veterinária</span> <br>
            <small class=\\"card-preco\\">R$ 150,00</small>
        </div>
    </div>
    "
`;
